featuretools/entityset/timedelta.py

```diff
@@ -101,67 +104,100 @@ def _check_unit_plural(cls, s):
             return s.lower()
         return s
 
+    def get_value(self, unit=None):
+        if unit is not None:
+            return self.times[unit]
+        elif len(self.times.values()) == 1:
+            return list(self.times.values())[0]
+        else:
+            return self.times
+
+    def get_units(self):
+        return list(self.times.keys())
+
     def get_unit_type(self):
-        if self.unit == "o":
+        all_units = self.get_units()
+        if self._Observations in all_units:
             return None
-        elif self.unit in self._absolute_units:
-            return pd.Timedelta(self.value, self.unit)
+        elif self.is_absolute() and self.has_multiple_units() is False:
+            return pd.Timedelta(self.times[all_units[0]], all_units[0])
         else:
-            unit = self.readable_unit.lower()
-            return relativedelta(**{unit: self.value})
-
-    def get_name(self):
-        unit = self.readable_unit
-        if self.readable_unit == "Weeks":
-            # divide to convert back
-            return "{} {}".format(self.value / self._convert_to_days["w"], unit)
-        if self.value == 1:
-            unit = self.make_singular(unit)
+            readable_times = self.lower_readable_times()
+            return relativedelta(**readable_times)
```

Comment:
Could you add an issue about supporting relativedelta explicitly and what is needed to support it? It seems like we can parse out unit and values and delta_obj to it?

Reply:
Created #744